lib/rubocop/cop/style/redundant_line_continuations.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoid redundant line continuations.
+      #
+      # @example
+      #   # bad
+      #   foo. \
+      #     bar
+      #   foo \
+      #     .bar \
+      #       .baz
+      #   foo&. \
+      #     bar
+      #   [foo, \
+      #     bar]
+      #   {foo: \
+      #     bar}
+      #   foo(bar, \
+      #     baz)
+      #
+      #   # good
+      #   foo.
+      #     bar
+      #   foo
+      #     .bar
+      #       .baz
+      #   foo&.
+      #     bar
+      #   [foo,
+      #     bar]
+      #   {foo:
+      #     bar}
+      #   foo(bar,
+      #     baz)
+      #   foo('bar' \
+      #     'baz')
+      #
+      class RedundantLineContinuations < Base
+        include MatchRange
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Avoid redundant line continuations.'
+
+        def on_send(node)
+          return unless redundant_line_continuation?(node)
+
+          register_offense(node.source_range) if node.dot? || node.safe_navigation?
+          register_offense(arguments_range(node)) if node.arguments?
+        end
+        alias on_csend on_send
+
+        def on_array(node)
+          return unless redundant_line_continuation?(node)
+
+          register_offense(node.source_range)
+        end
+        alias on_hash on_array
```

Comment:
Perhaps a more generic approach should be used rather than `on_send` and `on_array`.